pkg/util/kernel/download_headers.go

```diff
@@ -0,0 +1,123 @@
+// +build linux
+
+package kernel
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/nikos/apt"
+	"github.com/DataDog/nikos/cos"
+	"github.com/DataDog/nikos/rpm"
+	"github.com/DataDog/nikos/types"
+	"github.com/DataDog/nikos/wsl"
+)
+
+// customLogger is a wrapper around our logging utility which allows nikos to use our logging functions
+type customLogger struct{}
+
+func (c customLogger) Debug(args ...interface{})                 { log.Debug(args...) }
+func (c customLogger) Info(args ...interface{})                  { log.Info(args...) }
+func (c customLogger) Warn(args ...interface{})                  { log.Warn(args...) }
+func (c customLogger) Error(args ...interface{})                 { log.Error(args...) }
+func (c customLogger) Debugf(format string, args ...interface{}) { log.Debugf(format, args...) }
+func (c customLogger) Infof(format string, args ...interface{})  { log.Infof(format, args...) }
+func (c customLogger) Warnf(format string, args ...interface{})  { log.Warnf(format, args...) }
+func (c customLogger) Errorf(format string, args ...interface{}) { log.Errorf(format, args...) }
+
+var _ types.Logger = customLogger{}
+
+// downloadHeaders attempts to download kernel headers & place them in headerDownloadDir
+func downloadHeaders(headerDownloadDir string) ([]string, error) {
+	var (
+		target    types.Target
+		backend   types.Backend
+		outputDir string
+		err       error
+	)
+
+	if outputDir, err = createOutputDir(headerDownloadDir); err != nil {
+		return nil, fmt.Errorf("unable create output directory %s: %s", headerDownloadDir, err)
+	}
+
+	if target, err = getHeaderDownloadTarget(); err != nil {
+		return nil, fmt.Errorf("failed to retrieve target information: %s", err)
+	}
+
+	log.Infof("Downloading kernel headers for target distribution %s, release %s, kernel %s",
+		target.Distro.Display,
+		target.Distro.Release,
+		target.Uname.Kernel,
+	)
+	log.Debugf("Target OSRelease: %s", target.OSRelease)
+
+	if backend, err = getHeaderDownloadBackend(&target); err != nil {
+		return nil, fmt.Errorf("unable to get kernel header download backend: %s", err)
+	}
+
+	if err = backend.GetKernelHeaders(outputDir); err != nil {
+		return nil, fmt.Errorf("failed to download kernel headers: %s", err)
+	}
+
+	headerDirs := []string{fmt.Sprintf(filepath.Join(headerDownloadDir, kernelModulesPath), target.Uname.Kernel)}
+	switch strings.ToLower(target.Distro.Display) {
+	case "debian":
+		headerDirs = append(headerDirs, fmt.Sprintf(filepath.Join(headerDownloadDir, debKernelModulesPath), target.Uname.Kernel))
+	case "cos":
+		headerDirs = append(headerDirs, fmt.Sprintf(filepath.Join(headerDownloadDir, cosKernelModulesPath), target.Uname.Kernel))
+	}
```

Comment:
This logic seems to be similar to `getDefaultHeaderDirs`. Could we refactor to have it live in one place?